File: Algorithms/Sorting/Insertion_Sort/Insertion_Sort.py
# Python program for implementation of Insertion Sort 

def insertionSort(arr): 

	for i in range(1, len(arr)): 

		key = arr[i] 
		j = i-1
		while j >= 0 and key < arr[j] : 
				arr[j + 1] = arr[j] 
				j -= 1
		arr[j + 1] = key 


# Driver code to test above 

n=int(input("Enter number of elements: "))
print("Enter the elements: ")
arr = list(map(int,input().split())) 
insertionSort(arr) 
print("\nSorted array is: ")
for i in range(n): 
	print (arr[i], end=" ") 

File: Algorithms/Sorting/Insertion_Sort/insertion_sort.cs
using System; 
  
class InsertionSort { 

    // Insertion Sort Function
    void insertion_sort(int[] arr) 
    { 
        int array_Length = arr.Length; 
        for (int i = 1; i < array_Length; ++i) { 
            int key = arr[i]; 
            int j = i - 1; 
            
            while (j >= 0 && arr[j] > key) { 
                arr[j + 1] = arr[j]; 
                j = j - 1; 
            } 
            arr[j + 1] = key; 
        } 
    } 
    
    // Test Code 
    public static void Main() 
    { 
        //The Array That You Want To Sort
        int[] arr = { 9, 12, 11, 13, 5, 6, 99, 1, 2, 2, 4, 112, 44, 1}; 
        //Create Object To Call Function On
        InsertionSort obj = new InsertionSort(); 
        //Sort The Array
        obj.insertion_sort(arr); 
        //Prints Array
        Console.WriteLine(string.Join(" ", arr));
    } 
} 